src/repr/src/bytes.rs

```diff
@@ -0,0 +1,327 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::fmt::{self, Display};
+use std::str::FromStr;
+
+use proptest_derive::Arbitrary;
+use serde::{Deserialize, Serialize};
+
+use mz_ore::cast::CastLossy;
+
+/// Struct for postgres compatible size units which is different from
+/// `bytesize::ByteSize`. Instead of MiB or GiB and so on, it uses MB, GB for the sizes
+/// with 1024 multiplier.
+/// Valid units are B, kB, MB, GB, TB, PB with multiples of 1024
+/// where 1MB = 1024kB
+#[derive(
+    Arbitrary, Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize, Default,
+)]
+pub struct ByteSize {
+    size: u32,
```

Comment:
I changed this to `u32` from `u64`, because while parsing since the number can have fractions, I am using `f64` which can hold less number of bits than `u64`. But this also means we are limiting the size specified to `u32::MAX` (It does not seem that bad though, people can always leverage higher units to represent bigger numbers).

The other option might be to keep this `u64` here and use some Decimal library while parsing from string.

Reply:
Not applicable anymore, refactored to store the entire bytes as `u64`